### PR TITLE
feat(upgrade_to_v3): add migration for repository key for kube-prometheus-stack

### PR DESCRIPTION
**Pull request by sumo-drosiek**
Related to:

https://github.com/SumoLogic/sumologic-kubernetes-collection/pull/2651/files

### CODE REVIEW

File: src/go/cmd/update-collection-v3/migrations/kube-prometheus-stack-repository/migrate.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	Rest map[string]interface{} `yaml:",inline"`
+}
+
+type Image struct {
+	Repository *interface{}           `yaml:"repository,omitempty"`
+	Rest       map[string]interface{} `yaml:",inline"`
+}
+
+func Migrate(yamlV2 string) (yamlV3 string, err error) {
+	valuesV2, err := parseValues(yamlV2)
```
Comment (astencel-sumo) [inline src/go/cmd/update-collection-v3/migrations/kube-prometheus-stack-repository/migrate.go:64]: `valuesV2` should be `inputValues`

File: src/go/cmd/update-collection-v3/migrations/kube-prometheus-stack-repository/migrate.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+	if err != nil {
+		return "", fmt.Errorf("error parsing input yaml: %v", err)
+	}
+
+	valuesV3 := migrate(&valuesV2)
```
Comment (astencel-sumo) [inline src/go/cmd/update-collection-v3/migrations/kube-prometheus-stack-repository/migrate.go:69]: `valuesV3` should be `outputValues`

File: src/go/cmd/update-collection-v3/migrations/kube-state-metrics-collectors/migrate_test.go [hunk 3]
```diff
@@ -1,60 +0,0 @@
-package kubestatemetricscollectors
```
Comment (astencel-sumo) [inline src/go/cmd/update-collection-v3/migrations/kube-state-metrics-collectors/migrate_test.go:1]: Why are you removing this test?

**Review by astencel-sumo — CHANGES_REQUESTED** [review]
You should not need to change other migrations, `kube-state-metrics-collectors` in this case.

File: src/go/cmd/update-collection-v3/migrations/kube-state-metrics-collectors/migrate_test.go (hunk 3 above)
Comment (sumo-drosiek) [reply src/go/cmd/update-collection-v3/migrations/kube-state-metrics-collectors/migrate_test.go]: typo :grimacing:

**Pull request merged**